server/src/main/java/org/cloudfoundry/identity/uaa/zone/IdentityZoneEndpoints.java

```diff
@@ -333,6 +327,11 @@ public ResponseEntity<IdentityZone> deleteIdentityZone(@PathVariable String id)
             logger.debug("Zone - deleting id[" + id + "]");
             // make sure it exists
             IdentityZone zone = zoneDao.retrieveIgnoreActiveFlag(id);
+            //If identity zone exists , check if it is already ported
+            OrchestratorZoneEntity orchestratorZone = zoneDao.retrieveOrchestratorZoneByIdentityZoneId(id);
```

Comment:
Can you please confirm zoneDao.retrieveOrchestratorZoneByIdentityZoneId(id) returns expected response for cases below

1. if zone is not ported what will it return? Null or ZoneDoesNotExistsException. See this https://github.com/GESoftware-CF/uaa/blob/feature_handle_ported_instance_deletion/server/src/main/java/org/cloudfoundry/identity/uaa/zone/JdbcIdentityZoneProvisioning.java#L81-L87
2. if we get orchestratorZone=null then orchestratorZone.getOrchestratorZoneName() in line 332 might throw NullPointerException

Reply:
If zone is not present then it would throw ZoneDoesNotExistsException (Test case - testGetOrchIdentityZoneById_NotFound) . If the zone is present but not ported , the result will be a record since there is a left join with the identity_zone table. 

We will not get orchestratorZone=null since then we would get ZoneDoesNotExitsException. If the zone is present we will get the row with the value of identityZoneId but the zone_name will be null. The orchestrator_zone_name os not used except for checking if the instance is already ported or not

Reply:
consider one case [Without porting],
1. create new zone "uaa-zone" in CF using cf cs
2. delete zone "uaa-zone" in CF using cf delete-service

will this work and what will be the response to the caller?